core/services/encryption/EncryptionKeyManager.php

```diff
@@ -0,0 +1,272 @@
+     * @param bool   $overwrite                 - prevents accidental overwriting of an existing key which would be bad
+     * @return bool
+     * @throws Exception
+     */
+    public function addEncryptionKey($encryption_key_identifier, $encryption_key = '', $overwrite = false)
+    {
+        // ensure keys are loaded
+        $this->retrieveEncryptionKeys();
+        $encryption_key_identifier = $encryption_key_identifier ?: $this->default_encryption_key_id;
+        if (isset($this->encryption_keys[ $encryption_key_identifier ]) && ! $overwrite) {
+            // WOAH!!! that key already exists and we don't want to overwrite it
+            throw new RuntimeException (
+                sprintf(
+                    esc_html__(
+                        'The "%1$s" encryption key already exists and can not be overwritten because previously encrypted values would no longer be capable of being decrypted.',
+                        'event_espresso'
+                    ),
+                    $encryption_key_identifier
+                )
+            );
+        }
+        $this->encryption_keys[ $encryption_key_identifier ] = $encryption_key ?: $this->generateEncryptionKey();
+        return $this->saveEncryptionKeys();
+    }
+
+
+    /**
+     * returns cryptographically secure passphrase. will use default if necessary
+     *
+     * @param string $encryption_key_identifier - used for saving encryption key. will use default if necessary
+     * @param string $generate                  - will generate a new key if the requested one does not exist
+     * @return string
+     * @throws Exception
+     * @throws OutOfBoundsException
+     */
+    public function getEncryptionKey($encryption_key_identifier = '', $generate = true)
+    {
+        $encryption_key_identifier = $encryption_key_identifier ?: $this->default_encryption_key_id;
+        // ensure keys are loaded
+        $this->retrieveEncryptionKeys();
+        // if encryption key has not been set
+        if (! isset($this->encryption_keys[ $encryption_key_identifier ])) {
+            if ($generate) {
+                $this->addEncryptionKey($encryption_key_identifier);
+            } else {
+                throw new OutOfBoundsException(
+                    sprintf(
+                        esc_html__('The "%1$s" encryption key was not found or is invalid.', 'event_espresso'),
+                        $encryption_key_identifier
+                    )
+                );
+            }
+        }
+        return $this->encryption_keys[ $encryption_key_identifier ];
+    }
+
+
+    /**
+     * creates a new encryption key
+     *
+     * @param bool $strong if true (default) will attempt to generate a cryptographically secure key
+     * @return string
+     * @throws Exception
+     */
+    public function generateEncryptionKey($strong = true)
+    {
+        return $strong && PHP_VERSION_ID >= 70100
```

Comment:
```suggestion
     * @param string $type 'STRONG' or 'WEAK'. If 'STRONG' (default) will attempt to generate a cryptographically secure key
     * @return string
     * @throws Exception
     */
    public function generateEncryptionKey($type = 'STRONG')
    {
        return $type === 'STRONG' && PHP_VERSION_ID >= 70100
```

Reply:
??? 

is this for readability on the function calling side of things?

I get that argument, but prefer having PhpStorm tell me what those bools mean:

![ScreenShot_20210818104315](https://user-images.githubusercontent.com/1751030/129951734-06449a22-a9cc-47ed-a7d7-338ec1fe6a73.png)

Reply:
Yes it's for readability on the function calling side of things.

> I get that argument, but prefer having PhpStorm tell me what those bools mean

Cool